src/site/server/index.markdown

```diff
@@ -8,40 +8,168 @@ has-permalinks: false
 # {{ page.title }}
 
 Dart is a great fit for server applications.
-For example, Dart provides support for
-[futures](/docs/tutorials/futures) in the core
-[async](https://api.dartlang.org/apidocs/channels/stable/dartdoc-viewer/dart:asy
-nc)
-library, a robust [stream API](/docs/tutorials/streams), and
-[WebSockets](/samples/websockets/).
 
-### Where can I find more information about writing server code in Dart?
+* Runs nearly everywhere
+* Does nearly everything
+* Easy to develop and debug
+* Scales out
 
-Get started with [tutorials](/docs/tutorials), [articles](/articles/),
-and other documentation. Once you're coding, consult the
+The Dart VM runs anywhere you can get a Linux, Windows, or Mac prompt.
+It has been ported to i32, x64, ARM, ARM64, and MIPS.
+
+You can access files, UDP and TCP sockets, HTTP, Web sockets, and more with the
+[dart:io libraries][dart_io_libs]. Write a server, connect to a REST service,
+parse JSON, and more.
+
+Debugging and profiling your server app is easy with
+[Observatory](/tools/observatory/). Connect
+to your VM with your browser, and get real-time insight into your running
+app.
+
+Use [Isolates][isolate-docs] for secure, concurrent apps. Spawn an isolate to
+run Dart
+functions and libraries in an _isolated heap_, and take advantage of
+multiple CPU cores.
+
+Asynchronous code is easy to write, with the new _async/await/yield_
+language features.
+
+RPC services are easy to create and consume, with the
+[rpc package](https://github.com/dart-lang/rpc).
+You can write your RPC server in Dart, and generate client libraries for
+Java, C#, and more.
+
+### Examples
+
+Build an HTTP server:
+
+{% prettify dart %}
+import 'dart:io';
+
+main() async {
+  var server = await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 8080);
+  print("Serving at ${server.address}:${server.port}");
+  
+  await for (var request in server) {
+    request.response
+      ..headers.contentType = new ContentType("text", "plain", charset: "utf-8")
+      ..write('Hello, world')
+      ..close();
+  }
+}
+{% endprettify %}
+
+Server Web sockets:
+
+{% prettify dart %}
+import 'dart:io';
+import 'dart:async';
+
+handleMsg(msg) {
+  print('Message received: $msg');
+}
+
+main() {
+  runZoned(() async {
+    var server = await HttpServer.bind('127.0.0.1', 4040);
+    await for (var req in server) {
+      if (req.uri.path == '/ws') {
+        // Upgrade a HttpRequest to a WebSocket connection.
+        var socket = await WebSocketTransformer.upgrade(req);
+        socket.listen(handleMsg);
+      };
+    }
+  },
+  onError: (e) => print("An error occurred."));
+}
+{% endprettify %}
+
+Get HTTP resources:
+
+{% prettify dart %}
+import 'package:http/http.dart' as http;
+
+main() async {
+  var contents = await http.read('http://example.com/');
+  print(contents);
+}
+{% endprettify %}
+
+### Get started
+
```

Comment:
Perhaps add a link to the server codelab, https://www.dartlang.org/codelabs/server/ as well.